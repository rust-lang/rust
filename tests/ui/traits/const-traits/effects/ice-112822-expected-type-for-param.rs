#![feature(const_trait_impl)]

const fn test() -> impl ~const Fn() {
    //~^ ERROR `~const` can only be applied to `#[const_trait]` traits
    //~| ERROR `~const` can only be applied to `#[const_trait]` traits
    //~| ERROR `~const` can only be applied to `#[const_trait]` traits
    const move || { //~ ERROR const closures are experimental
        let sl: &[u8] = b"foo";

        match sl {
            [first, remainder @ ..] => {
                assert_eq!(first, &b'f');
                //~^ ERROR cannot call non-const fn
                //~| ERROR cannot call non-const operator
            }
            [] => panic!(),
        }
    }
}

fn main() {}
