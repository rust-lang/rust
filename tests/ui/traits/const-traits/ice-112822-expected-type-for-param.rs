#![feature(const_trait_impl)]

const fn test() -> impl [const] Fn() {
    //~^ ERROR: }: [const] Fn()` is not satisfied
    const move || { //~ ERROR const closures are experimental
        let sl: &[u8] = b"foo";

        match sl {
            [first, remainder @ ..] => {
                assert_eq!(first, &b'f');
                // FIXME(const_closures) ^ ERROR cannot call non-const function
            }
            [] => panic!(),
        }
    }
}

fn main() {}
