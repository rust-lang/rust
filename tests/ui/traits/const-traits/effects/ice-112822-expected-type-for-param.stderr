error[E0658]: const closures are experimental
  --> $DIR/ice-112822-expected-type-for-param.rs:7:5
   |
LL |     const move || {
   |     ^^^^^
   |
   = note: see issue #106003 <https://github.com/rust-lang/rust/issues/106003> for more information
   = help: add `#![feature(const_closures)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error: `~const` can only be applied to `#[const_trait]` traits
  --> $DIR/ice-112822-expected-type-for-param.rs:3:25
   |
LL | const fn test() -> impl ~const Fn() {
   |                         ^^^^^^

error: `~const` can only be applied to `#[const_trait]` traits
  --> $DIR/ice-112822-expected-type-for-param.rs:3:25
   |
LL | const fn test() -> impl ~const Fn() {
   |                         ^^^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error: `~const` can only be applied to `#[const_trait]` traits
  --> $DIR/ice-112822-expected-type-for-param.rs:3:25
   |
LL | const fn test() -> impl ~const Fn() {
   |                         ^^^^^^
   |
   = note: duplicate diagnostic emitted due to `-Z deduplicate-diagnostics=no`

error[E0015]: cannot call non-const operator in constant functions
  --> $DIR/ice-112822-expected-type-for-param.rs:12:17
   |
LL |                 assert_eq!(first, &b'f');
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants
   = note: this error originates in the macro `assert_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0015]: cannot call non-const fn `core::panicking::assert_failed::<&u8, &u8>` in constant functions
  --> $DIR/ice-112822-expected-type-for-param.rs:12:17
   |
LL |                 assert_eq!(first, &b'f');
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: calls in constant functions are limited to constant functions, tuple structs and tuple variants
   = note: this error originates in the macro `assert_eq` (in Nightly builds, run with -Z macro-backtrace for more info)

error: aborting due to 6 previous errors

Some errors have detailed explanations: E0015, E0658.
For more information about an error, try `rustc --explain E0015`.
